To: vim_dev@googlegroups.com
Subject: Patch 7.3.094
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.094
Problem:    Using abs() requires type cast to int.
Solution:   Use labs() so that the value remains long. (Hong Xu)
Files:	    src/screen.c


*** ../vim-7.3.093/src/screen.c	2010-12-08 19:56:52.000000000 +0100
--- src/screen.c	2010-12-30 14:54:08.000000000 +0100
***************
*** 2317,2323 ****
  		num = (long)lnum;
  	    else
  		/* 'relativenumber', don't use negative numbers */
! 		num = (long)abs((int)get_cursor_rel_lnum(wp, lnum));
  
  	    sprintf((char *)buf, "%*ld ", w, num);
  #ifdef FEAT_RIGHTLEFT
--- 2317,2323 ----
  		num = (long)lnum;
  	    else
  		/* 'relativenumber', don't use negative numbers */
! 		num = labs((long)get_cursor_rel_lnum(wp, lnum));
  
  	    sprintf((char *)buf, "%*ld ", w, num);
  #ifdef FEAT_RIGHTLEFT
***************
*** 3475,3482 ****
  			    num = (long)lnum;
  			else
  			    /* 'relativenumber', don't use negative numbers */
! 			    num = (long)abs((int)get_cursor_rel_lnum(wp,
! 								    lnum));
  
  			sprintf((char *)extra, "%*ld ",
  						number_width(wp), num);
--- 3475,3481 ----
  			    num = (long)lnum;
  			else
  			    /* 'relativenumber', don't use negative numbers */
! 			    num = labs((long)get_cursor_rel_lnum(wp, lnum));
  
  			sprintf((char *)extra, "%*ld ",
  						number_width(wp), num);
*** ../vim-7.3.093/src/version.c	2010-12-30 14:50:46.000000000 +0100
--- src/version.c	2010-12-30 14:56:32.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     94,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
58. You turn on your computer and turn off your wife.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
